### PR TITLE
docs: Add direnv usage

### PR DESCRIPTION
**Pull request by pavelzw**
https://github.com/prefix-dev/pixi/discussions/397#discussioncomment-8572062

I haven't tested this on windows

### CODE REVIEW

**Comment by pavelzw** [comment]
@ruben-arts I think something with your comment PR is wrong 

image

File: docs/environment.md [hunk 1]
```diff
@@ -86,6 +86,49 @@ It sets the `PATH` and some more environment variables. But more importantly it
...
+
+1. This ensures that your pixi environment is always up to date when you enter the directory.
+2. This activates the environment. `direnv` ensures that the environment is deactivated when you leave the directory.
+
+```shell
+$ cd my-project
+direnv: error /my-project/.envrc is blocked. Run `direnv allow` to approve its content
+$ direnv allow
+direnv: loading /my-project/.envrc
+✔ Project in /my-project is ready to use!
```
Comment (pavelzw) [inline docs/environment.md:122]: It is a bit annoying that this is always displayed when entering the directory. You can mitigate this by using `pixi install 2>/dev/null` but this leads to the installation loading bar to be hidden. Ideally there would be a cli flag `pixi install -q` that still displays a loading bar during installation but doesn't output the `Project in ... is ready to use!`

Wdyt?

Comment (baszalmstra) [reply docs/environment.md:122]: Wouldnt you expect a -q to also hide the progressbars?

Comment (pavelzw) [reply docs/environment.md:122]: Yes, that's the problem :/

Comment (baszalmstra) [reply docs/environment.md:122]: Alternatively we could print the text to stdout instead of stderr..?

**Comment by ruben-arts** [comment]
> @ruben-arts I think something with your comment PR is wrong

What do you think is the issue?

File: docs/environment.md (hunk 1 above)
Comment (pavelzw) [reply docs/environment.md:122]: When we skip the `pixi install` part, the message is only displayed when something actually changed 🤩

**Comment by pavelzw** [comment]
@ruben-arts FYI the docs deployment failed
image
https://github.com/prefix-dev/pixi/actions/runs/8046458199/job/21973710915
```
✨ Pixi task (docs): mike deploy --push dev devel
INFO    -  Cleaning site directory
INFO    -  Building documentation to directory: /home/runner/work/pixi/pixi/site
INFO    -  Documentation built in 0.8[7](https://github.com/prefix-dev/pixi/actions/runs/8046458199/job/21973710915#step:5:8) seconds
error: failed to push branch gh-pages to origin:
  remote: Permission to prefix-dev/pixi.git denied to github-actions[bot].
  fatal: unable to access 'https://github.com/prefix-dev/pixi/': The requested URL returned error: 403
Error: Process completed with exit code 1.
```

**Comment by ruben-arts** [comment]
I'm aware, its on my list ;) But thanks for pinging!

**Comment by ruben-arts** [comment]
It is uploaded now! And it should work on next push to main

**Pull request merged**